### PR TITLE
feat(fw): Remove `_make_request` and replace using it to `EthRPC` methods

### LINKED ISSUE #561: feat/refactor: extend `EthRPC` framework class and use it in `gentest.py`
#556 added classes to help calling ETH and Engine API methods on clients, we should extend these classes and refactor `gentest.py` to use them:
https://github.com/ethereum/execution-spec-tests/blob/aa1a57137084e4865651f6f780627c4a4e13a429/src/cli/gentest.py#L243-L254

### PR DESCRIPTION
**Pull request by artemd24**
## 🗒️ Description
Remove `_make_request` from `RequestManager` in `cli.gentest.py`. 
Replace using `_make_request` to next `EthRPC` methods: `get_block_by_number`, `get_transaction_by_hash`, `debug_trace_call`. 
Add 2 more new methods to `EthRPC`: `get_transaction_by_hash`, `debug_trace_call`.

Extend `EthRPC` add new methods:
1. `get_transaction_by_hash(self, transaction_hash: str)`
2. `debug_trace_call(self, tr: RequestManager.RemoteTransaction)`

## 🔗 Related Issues
Resolve #561

## ✅ Checklist
- [X] All: Set appropriate labels for the changes.
- [X] All: Considered squashing commits to improve commit history.
- [X] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [X] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.

### CODE REVIEW

**Comment by artemd24** [comment]
Hello @danceratopz,
please verify changes

File: src/ethereum_test_tools/rpc/rpc.py
Comment (danceratopz) [inline src/ethereum_test_tools/rpc/rpc.py]: This introduces a circular import. I think we should avoid using `RequestManager` objects here in rpc.py. 

The easiest way to resolve this would be to create a dict of the required arguments in `gentest` and have `debug_trace_call` take this dictionary as an argument. 

We could consider defining a small pydantic BaseModel of the required arguments and passing an instance of this model instead for better type-checking of the parameters, but I'm not sure it's worth it here.

File: src/ethereum_test_tools/rpc/rpc.py
Comment (danceratopz) [inline src/ethereum_test_tools/rpc/rpc.py:11]: This introduces a circular dependency, I think we should avoid using a `RequestManager` object here in rpc.py. 

The easiest solution would be to build a dict containing the required parameters and pass this dictionary to `debug_trace_call` as an argument.

We could consider defining a small pydantic BaseModel for these parameters, but it could be overkill in this case - a dict should be fine for now. But something we could consider adding later on.

**Review by danceratopz — COMMENTED** [review]
Thanks a lot for getting on this!

Whilst reviewing your code, I realized that `gentest` is currently broken, so you wouldn't have been able to execute your code. I've addressed this in https://github.com/ethereum/execution-spec-tests/pull/576.

It might be worth waiting for #576 to get merged before making heavier changes here, but in the meantime, you can try to get gentest (in the #576 branch) running, for this you'll need access to an archive node. If you don't have one personally, you can get it running using a free endpoint from https://www.portal.grove.city, for example. Check gentest module docstring in #576 for more help. 

Otherwise, we use tox to lint and type-check our code in order to enforce our code standards. Some of these are rather pedantic, but are generally very useful. Tox is executed as part of Github but you can run tox locally first to ensure that Github Actions pass. You I fixed up import ordering in 43f5e3ff7276187f1b20fb33fc1e7c3de15bfa4a, but we still have a few other issues, which I've noted below. Feel free to ping me for help (danceratopz on telegram/discord, if you get stuck). Here's some help to get started with:
- https://ethereum.github.io/execution-spec-tests/main/writing_tests/code_standards/
- https://ethereum.github.io/execution-spec-tests/main/writing_tests/verifying_changes/

File: src/ethereum_test_tools/rpc/rpc.py
Comment (artemd24) [reply src/ethereum_test_tools/rpc/rpc.py:11]: Ok, I'll try to tackle it

**Comment by artemd24** [comment]
@danceratopz 
Thank you for the detailed comment!
I agree with you to wait until #576 will be merged
however I'm going to add some fixes from your comment, it looks like there are some work that has to be done

File: src/ethereum_test_tools/rpc/rpc.py
Comment (spencer-tb) [inline src/ethereum_test_tools/rpc/rpc.py:132]: Very happy to keep this here, although the `EthRPC` class was to denote then `eth_` for calls. 

Could we move this function to its own class `DebugRPC`? To denote the `debug_` calls?

I'll do the same for the engine calls in the future with `EngineRPC`! 

Maybe for another time though

File: src/ethereum_test_tools/rpc/rpc.py
Comment (spencer-tb) [inline src/ethereum_test_tools/rpc/rpc.py:95]: It would be amazing if we could get he return type to match the `Transaction` pydantic type. Can also do this in the future, when refining the eth rpc classes!

**Review by spencer-tb — COMMENTED** [review]
Some quick comments on my side! Nice one :D

File: src/ethereum_test_tools/rpc/rpc.py
Comment (danceratopz) [reply src/ethereum_test_tools/rpc/rpc.py:132]: Oh good point.

File: docs/CHANGELOG.md
Comment (danceratopz) [reply docs/CHANGELOG.md]: Done :)

**Review by danceratopz — APPROVED** [review]
Hey @artemd24, thanks again for this! I'll be afk for a while and didn't want to leave this hanging while away, so I tried this out today and made a few fixes - I hope you don't mind. I tried to split the commits logically, so you can use them and as a review. I also asked @spencer-tb for some quick feedback :slightly_smiling_face:.

Feel free to pick up on @spencer-tb's comments and work on any of them them in a new PR!

File: src/cli/gentest.py
Comment (danceratopz) [reply src/cli/gentest.py]: Saving for a later PR, cc @artemd24.

File: src/ethereum_test_tools/rpc/rpc.py
Comment (danceratopz) [reply src/ethereum_test_tools/rpc/rpc.py:134]: Saving for a later PR, cc @artemd24.

**Comment by artemd24** [comment]
@danceratopz @spencer-tb 
Thank you very much for helping me! I was a bit stuck and didn't see where to go.
Anyway @danceratopz thank you for your patient and for refactoring I will certainly examine all changes!
Additional thanks to @spencer-tb, your comments are very useful for me in perspective of improvement of my involvement and contribution in opensource

**Pull request merged**
